pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -63,7 +63,7 @@
      */
     byte[] getData();
 
-    T getValue();
+    T getValue() throws SchemaSerializationException;
```

Comment:
Why add the exception when getting the value? I think the schema should be validated internally: if a messages reaches the application is should already be "valid" and not throw exception here, because that would force the application to handle it and 90% of people would be confused on what to do at this point.